lib/active_record/connection_adapters/duckdb/database_statements.rb

```diff
@@ -3,40 +3,289 @@
+        # @note DuckDB-specific query execution
+        # @param [Object] raw_connection Raw database connection
+        # @param [String] sql SQL to execute
+        # @param [Array] binds Bind parameters
+        # @param [Array] type_casted_binds Type-casted bind parameters
+        # @param [Boolean] prepare Whether to prepare statement
+        # @param [Object] notification_payload Notification payload for logging
+        # @param [Boolean] batch Whether to execute in batch mode
+        # @return [Object] Query result
+        def perform_query(raw_connection, sql, binds, type_casted_binds, prepare:, notification_payload:, batch: false)
+          # Use DuckDB's native parameter binding - clean and secure
+          bind_values = extract_bind_values(type_casted_binds, binds)
+          
+          if bind_values&.any?
+            @raw_connection.query(sql, *bind_values)
+          else
+            @raw_connection.query(sql)
+          end
+        end
 
-          # TODO: https://github.com/suketa/ruby-duckdb/issues/168
-          # build_result(columns: result.columns, rows: result.to_a)
-          if result.to_a.first&.size == 1
-            build_result(columns: ['count'], rows: result.to_a)
-          elsif result.to_a.first&.size == 2
-            build_result(columns: ['id', 'name'], rows: result.to_a)
+        # @override
+        # @note Implements AbstractAdapter interface method
+        # @param [String] sql SQL to execute
+        # @param [String, nil] name Query name for logging
+        # @return [Object] Query result
+        def query(sql, name = nil)
+          result = internal_execute(sql, name)
+          result
+        end
+
+        # @override
+        # @note Implements AbstractAdapter interface method
+        # @param [String] sql SQL to explain
+        # @return [String] Pretty-printed explanation
+        def explain(sql)
+          result = internal_exec_query(sql, "EXPLAIN")
+          Duckdb::ExplainPrettyPrinter.new.pp(result)
+        end
+
+        # @override
+        # @note Implements AbstractAdapter interface method - Executes an INSERT statement and returns the ID of the newly inserted record
+        # @param [String] sql INSERT SQL to execute
+        # @param [String, nil] name Query name for logging
+        # @param [Array] binds Bind parameters
+        # @param [String, nil] pk Primary key column name
+        # @param [String, nil] sequence_name Sequence name for auto-increment
+        # @param [String, nil] returning RETURNING clause
+        # @return [ActiveRecord::Result] Result containing inserted ID
+        def exec_insert(sql, name = nil, binds = [], pk = nil, sequence_name = nil, returning: nil)
+          if pk && supports_insert_returning?
+            # Use INSERT...RETURNING to get the inserted ID
+            returning_sql = sql.sub(/\bINSERT\b/i, "INSERT").concat(" RETURNING #{quote_column_name(pk)}")
```

Comment:
It seems that we don't need the `sub` here:

```suggestion
            returning_sql = "#{sql} RETURNING #{quote_column_name(pk)}"
```

BTW, do we need to override `exec_insert`? It seems that we can use the default `sql_for_insert` and `exec_insert` implementations in Active Record.